### PR TITLE
Handle loading screens

### PR DESCRIPTION
**Pull request by kozaktar**
-Fixed bug where onboarding page would briefly appear when logging in. Now it displays loading screen instead.
-Added withSpinner component to make it easy to add lading page to any components.
Usage: (adding spinner to Modal component)

const ModalWithSpinner=WithSpinner(Modal);
 <ModalWithSpinner isloading={loadingState}/>
 
If loadingState is set to true loading indicator will be displayed instead of the component. If loadingState is set to false the component will be displayed as normal.

### CODE REVIEW

File: client/src/components/WithSpinner.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import LoadingScreen from "./LoadingScrean";
```
Comment (dproc96) [inline client/src/components/WithSpinner.js:2]: Can you run prettier on this file?

File: client/src/pages/OnBoarding.js [hunk 2]
```diff
@@ -46,7 +50,8 @@ const OnBoarding = () => {
 
   return (
     <>
-      {onboarded ? (
+      {
+        loading?(<LoadingScreen/>):onboarded ? (
```
Comment (dproc96) [inline client/src/pages/OnBoarding.js]: Can you run prettier on this file as well?

**Review by dproc96 — APPROVED** [review]
Pending some code formatting, this is good to go

**Comment by dproc96** [comment]
Actually hold off on merging this until Matt's changes have been merged in

File: client/src/components/LoadingScrean.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { makeStyles } from '@material-ui/core/styles';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+const useStyles = makeStyles((theme) => ({
```
Comment (mattchx) [inline client/src/components/LoadingScrean.js]: Good Styling!

File: server/routes/teamEventAPI.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+const express = require("express");
```
Comment (mattchx) [inline server/routes/teamEventAPI.js]: Did you add this in by accident ?

**Review by mattchx — CHANGES_REQUESTED** [review]
I'd suggest removing teamEventAPI.js

File: server/routes/teamEventAPI.js (hunk 4 above)
Comment (kozaktar) [reply server/routes/teamEventAPI.js]: I don't even know how this got here. I don't remember changing any backend files....
Maybe I merged dev into this branch before creating the pull request?

**Pull request merged**